src/raytracer/xmlreader.m

```diff
@@ -63,114 +65,149 @@
 end
 %% Iterating through all the subdivisions (volumes) and extracting the triangle information
 
-volume=s.amf.object.mesh.volume';
-countRows=1;
-sizeVolume=size(volume);
-for iterateVolume=1:size(volume)
-    if sizeVolume(1)~=1
-        triangles=s.amf.object.mesh.volume{1,iterateVolume}.triangle';
+% countRows=1; 
+CADOutput = [];
+objLength=length(s.amf.object);                                                            
+for iobjLength = 1:objLength                            % For multiple objects          
+    if objLength == 1
+        volume=s.amf.object.mesh.volume';                                     
+        sizeVolume=size(volume);
+        stemp = s.amf.object;                                                 
     else
-        triangles=s.amf.object.mesh.volume.triangle';
+        volume=s.amf.object{1,iobjLength}.mesh.volume';                                     
+        sizeVolume=size(volume);
+        stemp = s.amf.object{1,iobjLength};                                                 
     end
-    
-    if materialSwitch==1
+    for iterateVolume=1:sizeVolume
         if sizeVolume(1)~=1
-            materialid=s.amf.object.mesh.volume{1,iterateVolume}.Attributes.materialid;
+            triangles=stemp.mesh.volume{1,iterateVolume}.triangle';                     
         else
-            materialid=s.amf.object.mesh.volume.Attributes.materialid;
+            triangles=stemp.mesh.volume.triangle';                                      
         end
-        
-        for iterateMaterials=1:sizeMaterials
-            if sizeMaterials~=1
-                if str2double(materialid)==str2double(s.amf.material{1,iterateMaterials}.Attributes.id)
-                    Material=s.amf.material{1,iterateMaterials}.metadata.Text;
-                end
-                
-            elseif sizeVolume(1)==1 && sizeMaterials == 1
-                if str2double(materialid)==str2double(s.amf.material.Attributes.id)
-                    Material=s.amf.material.metadata.Text;
+
+        if materialSwitch==1
+            if sizeVolume(1)~=1
+                materialid=stemp.mesh.volume{1,iterateVolume}.Attributes.materialid;   
+            else
+                materialid=stemp.mesh.volume.Attributes.materialid;                     
+            end
+
+            for iterateMaterials=1:sizeMaterials
+                if sizeMaterials~=1
+                    if str2double(materialid)==str2double(s.amf.material{1,iterateMaterials}.Attributes.id)
+                        Material=s.amf.material{1,iterateMaterials}.metadata.Text;
+                    end
+
+                elseif sizeVolume(1)==1 && sizeMaterials == 1
+                    if str2double(materialid)==str2double(s.amf.material.Attributes.id)
+                        Material=s.amf.material.metadata.Text;
```

Comment:
you could also `break` after this, right?

Reply:
Did not get this comment. please elaborate

Reply:
if you find the material you don't need to check the others, so you can `break` the for loop